### PR TITLE
I18n

### PR DESCRIPTION
**Pull request by borisyankov**
Based on 'themes' pull request

* Adds support for internationalization
* Uses ract-intl
* Files used as downloaded directly from Transifex
* All components that display any text are react-intl aware

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@borisyankov, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/common/ErrorMsg.js [hunk 1]
```diff
@@ -27,7 +29,9 @@ export default class ErrorMsg extends React.PureComponent {
 
     return (
       <View style={styles.field}>
-        <Text style={styles.error}>{error}</Text>
+        <Label style={styles.error}>
+          {error}
```
Comment (kunall17) [inline src/common/ErrorMsg.js]: Label not having text ultimately FormattedMessage not having ID throwing an error

File: src/start/AuthScreen.js [hunk 2]
```diff
@@ -31,11 +31,8 @@ class AuthScreen extends React.Component {
           keyboardShouldPersistTaps="always"
         >
           <View style={styles.container}>
-            <Text style={[styles.field, styles.heading1]}>
-              Welcome to Zulip!
-            </Text>
-            <Input
-              customStyle={[styles.field, styles.disabled]}
+            <Label
+              style={[styles.field]}
```
Comment (kunall17) [inline src/start/AuthScreen.js]: Label not having text ultimately FormattedMessage not having ID throwing an error

**Comment by borisyankov** [comment]
@kunall17 @Sam1301 does any of you have access to Transifex? Would be nice to translate the UI to Hindi. It is very few strings right now. I'll add the missing ones.

**Comment by kunall17** [comment]
Nope I don't have it, I requested a translation for Hindi btw!

**Comment by borisyankov** [comment]
I'll ask Tim to add you both. 
Just note, that the `translations/*.json` files are exported from Transifex and should not be edited manually.

**Comment by kunall17** [comment]
For the login part it throws '[React Intl] An `id` must be provided to format a message.' rest seems nice

**Pull request merged**